src/mainloop.rkt

```diff
@@ -396,15 +396,37 @@
   (extract!))
 
 (define (pareto-regimes sorted ctx)
-  (let loop ([alts sorted])
+  (define branch-exprs
+    (if (flag-set? 'reduce 'branch-expressions)
+        (exprs-to-branch-on sorted ctx)
+        (context-vars ctx)))
+  (define err-lsts (batch-errors (map alt-expr sorted) (*pcontext*) ctx))
+  (define init-errs (for/hash ([bexpr branch-exprs]) (values bexpr -1)))
+  (let loop ([alts sorted] [errs init-errs] [try-first (first branch-exprs)])
     (cond
      [(null? alts) '()]
      [(= (length alts) 1) (list (car alts))]
      [else
-      (define opt (infer-splitpoints alts ctx))
+      (define recomputed-branch-exprs
+        (if (flag-set? 'reduce 'branch-expressions)
+            (exprs-to-branch-on alts ctx)
+            (context-vars ctx)))
+      
+      (define recomp-errs 
+        (for/hash ([bexpr recomputed-branch-exprs]) (values bexpr (hash-ref errs bexpr))))
```

Comment:
Is it ever possible that we get *more* bexprs from having fewer alts? If yes, can't the `hash-ref` crash?

Reply:
I don't think this can happen and I can't think of a scenario where this would happen but we should keep an eye out for this